### DISCUSSION TITLE
DataFromArrayResolver suggestion

### DISCUSSION THREAD

**Discussion by Ashk2a** [spatie/laravel-data#67] [Ideas]
Hi,

I see isn't possible to use default value in __construct with data built from an array.
Example of code to trigger the error : 
```php
<?php

class ExampleData extends Data {
   public function __construct(public string $string, public bool $defaultBool = true) {}
}

$array = ['string'  => 'String example'];
$data = ExampleData::from($array); // Error throwing $defaultBool is null
```

My suggestion is change to change [this line](https://github.com/spatie/laravel-data/blob/main/src/Resolvers/DataFromArrayResolver.php#L100)

```php
// Current
return new $class(...$properties);

// Suggestion
return new $class(...$properties->filter());
```

What do you think?

**Comment by rubenvanassche** [comment]
Hey @Ashk2a,

The solution was a it more complex than you proposed (since default values can also be explicitly null) but is fixed in version 1.4.0.
